File: 056_Merge_Intervals/056_1.py
class Interval:
    def __init__(self, s=0, e=0):
        self.start = s
        self.end = e

class Solution:
    def merge(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: List[Interval]
        """
        if len(intervals) <= 1:
            return intervals
        intervals = sorted(intervals, key=lambda x : x.start)
        res = []
        init = intervals[0]
        for array in intervals[1:]:
            if init.end < array.start:
                res.append(init)
                init = array
            else:
                init.end = max(init.end, array.end)
        res.append(init)
        return res

File: 056_Merge_Intervals/056_1.java
public class Interval
{
    int start;
    int end;
    Interval() { start = 0; end = 0; }
    Interval(int s, int e) { start = s; end = e; }
}

class Solution {
    public List<Interval> merge(List<Interval> intervals) {
        if (intervals.size() <= 1)
            return intervals;

        intervals.sort((i1, i2) -> Integer.compare(i1.start, i2.start));
        List<Interval> ans = new ArrayList<>();
        Interval init = intervals.get(0);
        for (Interval array : intervals)
        {
            if (init.end < array.start)
            {
                ans.add(init);
                init = array;
            }
            else
            { init.end = Math.max(init.end, array.end); }
        }
        ans.add(init);
        return ans;
    }
}